File: components/redis-handling-session/src/main/java/com/sv/youapp/component/authorization/converter/UsernamePasswordAuthenticationTokenToBytesConverter.java
package com.sv.youapp.component.authorization.converter;

import org.springframework.core.convert.converter.Converter;
import org.springframework.data.convert.WritingConverter;
import org.springframework.data.redis.serializer.Jackson2JsonRedisSerializer;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;

@WritingConverter
public class UsernamePasswordAuthenticationTokenToBytesConverter
	implements Converter<UsernamePasswordAuthenticationToken, byte[]> {

	private final Jackson2JsonRedisSerializer<UsernamePasswordAuthenticationToken> serializer;

	public UsernamePasswordAuthenticationTokenToBytesConverter(
		Jackson2JsonRedisSerializer<UsernamePasswordAuthenticationToken> serializer
	) {
		this.serializer = serializer;
	}

	@Override
	public byte[] convert(UsernamePasswordAuthenticationToken value) {
		return this.serializer.serialize(value);
	}
}

File: platform-instances/authorization-server/src/main/kotlin/com/sv/youapp/service/authorization/converter/UsernamePasswordAuthenticationTokenToBytesConverter.kt
package com.sv.youapp.service.authorization.converter

import org.springframework.core.convert.converter.Converter
import org.springframework.data.convert.WritingConverter
import org.springframework.data.redis.serializer.Jackson2JsonRedisSerializer
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken

@WritingConverter
class UsernamePasswordAuthenticationTokenToBytesConverter(val serializer: Jackson2JsonRedisSerializer<UsernamePasswordAuthenticationToken?>) :
    Converter<UsernamePasswordAuthenticationToken?, ByteArray?> {
    override fun convert(value: UsernamePasswordAuthenticationToken): ByteArray {
        return this.serializer.serialize(value)
    }
}
